app/models/manageiq/providers/azure/cloud_manager/refresh_parser.rb

```diff
@@ -120,73 +123,78 @@ def process_collection(collection, key)
       end
 
       def parse_series(s)
-        name = uid = s['name']
+        name = uid = s.name
         new_result = {
           :type           => "ManageIQ::Providers::Azure::CloudManager::Flavor",
           :ems_ref        => uid,
           :name           => name,
-          :cpus           => s['numberOfCores'], # where are the virtual CPUs??
-          :cpu_cores      => s['numberOfCores'],
-          :memory         => s['memoryInMB'].to_f,
-          :root_disk_size => s['osDiskSizeInMB'] * 1024,
-          :swap_disk_size => s['resourceDiskSizeInMB'] * 1024
+          :cpus           => s.number_of_cores, # where are the virtual CPUs??
+          :cpu_cores      => s.number_of_cores,
+          :memory         => s.memory_in_mb.to_f,
+          :root_disk_size => s.os_disk_size_in_mb * 1024,
+          :swap_disk_size => s.resource_disk_size_in_mb * 1024
         }
         return uid, new_result
       end
 
       def parse_az(az)
-        id = az["id"].downcase
+        id = az.id.downcase
 
         new_result = {
           :type    => "ManageIQ::Providers::Azure::CloudManager::AvailabilityZone",
           :ems_ref => id,
-          :name    => az["name"],
+          :name    => az.name,
         }
         return id, new_result
       end
 
       def parse_cloud_network(cloud_network)
         cloud_subnets = get_cloud_subnets(cloud_network).collect do |raw_subnet|
-          @data_index.fetch_path(:cloud_subnets, raw_subnet.fetch_path('id'))
+          @data_index.fetch_path(:cloud_subnets, raw_subnet.id)
         end
 
-        uid = cloud_network.fetch_path('id')
+        uid = resource_uid(@subscription_id,
+                           cloud_network.resource_group.downcase,
+                           cloud_network.type.downcase,
+                           cloud_network.name)
+
```

Comment:
has to properly construct the `uid` so that it can be used to find the correct `orchestration_stack` association